pinot-spi/src/main/java/org/apache/pinot/spi/data/Schema.java

```diff
@@ -520,14 +535,60 @@ public SchemaBuilder setSchemaName(String schemaName) {
       return this;
     }
 
+    public SchemaBuilder withOptions(Consumer<Options> consumer) {
+      consumer.accept(_schema._options);
+      return this;
+    }
+
+    public SchemaBuilder addMetricField(String name, DataType dataType) {
+      return addMetricField(name, dataType, ignore -> {
+      });
+    }
+
+    public SchemaBuilder addMetricField(String name, DataType dataType, Consumer<MetricFieldSpec> customizer) {
```

Comment:
I don't see much value for adding this flexible customizer. `addField(FieldSpec fieldSpec)` can be handy though. This customizer is more or less creating a `FieldSpec`

Reply:
It is very valuable in order to keep the fluent dsl of the builder. There are several examples in this PR. Basically it is quite easier to write something like:

``` 
Schema schema = new SchemaBuilder()
   .addMetricField("name", type, field -> {
      //do your changes here
   })
   .build();
```

Than:

``` 
FieldSpec metric = new MetricField("name", type);
// do your changes here
Schema schema = new SchemaBuilder()
   .addField(field)
.build();
```

That is clear by the fact we have several builder methods like:
```
 public SchemaBuilder addSingleValueDimension(String dimensionName, DataType dataType);
 public SchemaBuilder addSingleValueDimension(String dimensionName, DataType dataType, Object defaultNullValue);
public SchemaBuilder addSingleValueDimension(String dimensionName, DataType dataType, int maxLength,
        Object defaultNullValue);
///etc
```

The current method does not scale while having a consumer is cleaner, shorter, scalable and safer. The later comes from the fact that having so many overloaded methods end up being problematic once we have a collision in types, while the consumer method is always type safe and explicit given we use named setters.